cms/bundles/models.py

```diff
@@ -0,0 +1,199 @@
+        """Provides a pre-filtered queryset for editable bundles. Usage: Bundle.objects.editable()."""
+        return self.filter(status__in=EDITABLE_BUNDLE_STATUSES)
+
+
+# note: mypy doesn't cope with dynamic base classes and fails with:
+# 'Unsupported dynamic base class "models.Manager.from_queryset"  [misc]'
+# @see https://github.com/python/mypy/issues/2477
+class BundleManager(models.Manager.from_queryset(BundlesQuerySet)):  # type: ignore[misc]
+    def get_queryset(self) -> BundlesQuerySet:
+        """Augments the queryset to order it by the publication date, then name, then reverse id."""
+        queryset: BundlesQuerySet = super().get_queryset()
+        queryset = queryset.alias(
+            release_date=Coalesce("publication_date", "release_calendar_page__release_date")
+        ).order_by(F("release_date").desc(nulls_last=True), "name", "-pk")
+        return queryset  # note: not returning directly to placate no-any-return
+
+
+class Bundle(index.Indexed, ClusterableModel, models.Model):  # type: ignore[django-manager-missing]
+    base_form_class = BundleAdminForm
+
+    name = models.CharField(max_length=255)
+    created_at = models.DateTimeField(auto_now_add=True)
+    created_by = models.ForeignKey(
+        "users.User",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="bundles",
+    )
+    # See https://docs.wagtail.org/en/stable/advanced_topics/reference_index.html
+    created_by.wagtail_reference_index_ignore = True  # type: ignore[attr-defined]
+
+    approved_at = models.DateTimeField(blank=True, null=True)
+    approved_by = models.ForeignKey(
+        "users.User",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="approved_bundles",
+    )
+    approved_by.wagtail_reference_index_ignore = True  # type: ignore[attr-defined]
+
+    publication_date = models.DateTimeField(blank=True, null=True)
+    release_calendar_page = models.ForeignKey(
+        "release_calendar.ReleaseCalendarPage",
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+        related_name="bundles",
+    )
+    status = models.CharField(choices=BundleStatus.choices, default=BundleStatus.PENDING, max_length=32)
+
+    objects = BundleManager()
+
+    panels: ClassVar[list["Panel"]] = [
+        FieldPanel("name"),
+        FieldRowPanel(
+            [
+                FieldPanel("release_calendar_page", heading="Release Calendar page"),
+                FieldPanel("publication_date", heading="or Publication date"),
+            ],
+            heading=_("Scheduling"),
+            icon="calendar",
+        ),
+        FieldPanel("status"),
+        InlinePanel("bundled_pages", heading=_("Bundled pages"), icon="doc-empty", label=_("Page")),
+        # these are handled by the form
+        FieldPanel("approved_by", classname="hidden w-hidden"),
+        FieldPanel("approved_at", classname="hidden w-hidden"),
+    ]
```

Comment:
This can be included in a future PR. Currently, users can select both "Live" release calendar pages and analysis pages. We need a way to restrict the selection to only pages that have not been published yet. This will likely require a custom chooser, unless the existing choosers already have this functionality. A simpler approach would be to validate the selection upon saving, notifying users that they cannot choose "Live" pages; however, this is not ideal for user experience.

The bundle chooser widget currently handles it, i.e. we override the queryset to only return "editable" bundles.

Reply:
We'll need to discuss this as it is more nuanced.
Added to my bundle DX ticket - https://jira.ons.gov.uk/browse/CMS-232